resources/assets/js/core/components/typeahead.vue

```diff
@@ -108,11 +141,52 @@ export default {
                 }
             }
         },
+        handleArrowKeyScroll(e) {
+            if (this.scrollable && this.items.length > 0) {
+                const typeahead = this.$refs.typeahead;
+                const scrollAmount = this.$refs.dropdown[0].$el.clientHeight;
+
+                // Reset scroll top if selected item is hidden
+                if (typeahead.scrollTop != this.selectedItemIndex * scrollAmount) {
+                    typeahead.scrollTop = this.selectedItemIndex * scrollAmount;
+                }
+
+                if (e.key === 'ArrowUp' && typeahead.scrollTop >= scrollAmount && this.selectedItemIndex > 0) {
+                    typeahead.scrollTop -= scrollAmount;
+                    this.selectedItemIndex -= 1;
+                }
+
+                if (e.key === 'ArrowDown' && typeahead.scrollTop < typeahead.scrollHeight && this.selectedItemIndex < this.maxItemCount - 1) {
+                    typeahead.scrollTop += scrollAmount;
+                    this.selectedItemIndex += 1;
+                }
+            }
+        }
     },
     watch: {
         value(value) {
             this.internalValue = value;
         },
+        inputText(v) {
+            this.isTyping = true;
+            clearTimeout(this.timerTask);
+
+            this.timerTask = setTimeout(() => {
+                let added = v.trim().includes(this.oldInput.trim());
+                let useTypeaheadFilter = this.oldInput.length > 3 && added;
+                if (v.length >= 3 && !useTypeaheadFilter) {
+                    this.$emit('fetch', v);
+                }
+                this.isTyping = false;
+                this.oldInput = v
+            }, 500);
```

Comment:
You can also use the [debounce](https://github.com/biigle/core/blob/1365b19497f1b57e3a2ba24374a78dfc7f606638/resources/assets/js/core/utils.js#L17-L22) helper method for this:

```js
debounce(() => {...}, 500, 'typeahead-fetch');
```